tool/menu.lua

```diff
@@ -148,36 +148,57 @@ end
 
 local function randomizermenu()
   local selected = ui.select({
-    "Proficiency: " .. randomizer.proficiency,
-    "Unseeded",
-    "Seeded",
+    "New Seed on New Game: " .. randomizer.newgamenewseedui[randomizer.newgamenewseed],
+    "Difficulty: " .. randomizer.queue[2].difficulty,
+    "Random seed",
+    "Set seed",
     "Reset",
     "Back",
   })
-  if selected == 1 then
-    local index = table.indexof(randomizer.proficiencies, randomizer.proficiency)
-    index = ((index - 1 + 1) % #randomizer.proficiencies) + 1
-    randomizer.proficiency = randomizer.proficiencies[index]
-  elseif selected == 2 then
-    randomizer.seed = ""
-    randomizer.randomize()
-    state = STATES.RANDOMIZER
-  elseif selected == 3 then
+  if selected == 1 then -- New Seed on New Game
+    local index = table.indexof(randomizer.newgamenewseedvalues, randomizer.newgamenewseed)
+    index = ((index - 1 + 1) % #randomizer.newgamenewseedvalues) + 1
+    randomizer.newgamenewseed = randomizer.newgamenewseedvalues[index]
+    if randomizer.newgamenewseed == "On" and #randomizer.queue == 2 then
+      randomizer.queue = {randomizer.queue[1], {seed = "", seedtype = randomizer.SEEDTYPE.RANDOMSEED, difficulty = randomizer.queue[2].difficulty}}
```

Comment:
I think we should move this statement into a function on `randomizer`. It will make reading here easier. Especially because `randomizer.queue[2]` must always exist and have some values defined, I think the queue should become an implementation detail of the `randomizer.lua` file and not ever be accessed from the outside. Therefore, it should become a `local` in there. The function(s) could look something like this
```lua
function randomizer.setnext(seedtype, seed, difficulty)
    seed = seed or ""
    difficulty = difficulty or randomizer.queue[2].difficulty
    -- set stuff
end
```